### PR TITLE
Option prerequisite

### PR DESCRIPTION
**Pull request by OzoneH3**
Implements #23183

### CODE REVIEW

**Comment by Leland** [comment]
So much yes. This works great.

Few notes:

1. "Safe-mode," "Safe mode proximity distance," et al are, in the PR, set to have "Auto-safe mode" as their prerequisite. This isn't correct – "Safe mode" is the parent for all of those. Will admit, the ordering in the settings is a bit screwy. (Open PR #23184 fixes that, putting "Safe mode" first.)

2. "Choose tileset" could have "Use tiles" as a prereq

3. "Software framebuffer acceleration" could have "Software rendering" as a prereq

File: src/options.h [hunk 1]
```diff
@@ -95,6 +95,9 @@ class options_manager
                     return !operator==( rhs );
                 }
 
+                void setPrerequisite( const std::string &sOption );
+                bool hasPrerequisite();
```
Comment (BevapDin) [inline src/options.h]: Should be `const`.

**Comment by Wulfe813** [comment]
@OzoneH3 You mentioned being able to streamline option descriptions in the issue. Naming the prerequisite option in the pop-up message would be a good thing(and more user friendly).

**Comment by OzoneH3** [comment]
Even though you can toggle safemode ingame, it's still the right prerequisite for the other safemode options. Fixed it so it's now on top of the list as well.

**Comment by Leland** [comment]
> Even though you can toggle safemode ingame, it's still the right prerequisite for the other safemode options

I think you might be right regarding the intent behind the option, but consider the following scenario:

1. Safe mode is set to False. All proceeding options are disabled.
2. Game is started. Safe mode is not on.
3. User changes safe mode to on via key command and wants to adjust the proximity in which safe mode is triggered

In this scenario, they would not be able to adjust the proximity. The "Safe mode" option, the prerequisite of Proximity, is not automatically toggled to True when the key command is hit.

Frankly, I think the option and all code relating to it should be removed. It's quite misleading.

**Comment by OzoneH3** [comment]
Ah I see now. I thought it would be toggled on in the options as well if you press !.
Would a new player toggle the safemode ingame though?
He wouldn't really know that ! toggles the safemode because it has to be ON to display the: Press ! to disable.
Kinda tricky.

**Comment by Leland** [comment]
> Would a new player toggle the safemode ingame though?
> He wouldn't really know that ! toggles the safemode because it has to be ON to display the: Press ! to disable.

A new user wouldn't, no – but the fact that it's possible is bad. For this PR, I'd probably just say leave off that prerequisite. You could also rename the option to, like, "Start games with safe mode activated"

I'll most likely be removing the option in a future PR. It seems worthless and (more importantly) confusing.

**Comment by OzoneH3** [comment]
Rebased to resolve conflicts and removed the SAFEMOD prerequisites.

**Pull request merged**